### PR TITLE
Upgrade golang to 1.12.8 due to multiple CVEs

### PR DESCRIPTION
**Pull request by mvladev**
**What this PR does / why we need it**:

The apiserver is using http2 and it can be target of DOS.

See:
https://groups.google.com/forum/#!topic/golang-nuts/fCQWxqxP8aA
https://github.com/golang/go/issues/33606

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Release note**:

```noteworthy developer
The golang version has been updated to `1.12.8` (fixing `CVE-2019-9512` and `CVE-2019-9514`).
```

/cc @ThormaehlenFred @ccwienk

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm

**Comment by mvladev** [comment]
Other components in the gardener project are not using HTTP2 so upgrade for them is not required

**Comment by rfranzke** [comment]
Thanks, Martin!

**Comment by mvladev** [comment]
Issue in K8S:

https://github.com/kubernetes/kubernetes/issues/79912

File: .ci/test [hunk 1]
```diff
@@ -34,7 +34,7 @@ fi
 cd "${SOURCE_PATH}"
 
 # Build the Ginkgo (test framework) binary to be able to execute the tests.
-GO111MODULE=on go install -mod=vendor ./vendor/github.com/onsi/ginkgo/ginkgo
+GO111MODULE=on GOFLAGS="-mod=vendor" go install ./vendor/github.com/onsi/ginkgo/ginkgo
```
Comment (ialidzhikov) [inline .ci/test]: Now half of the occurrences use `GOFLAGS="-mod=vendor"`, and the other half - `-mod=vendor`. Let's pick one and use only it.

Comment (rfranzke) [reply .ci/test]: +1, please make it consistent. Let's wait with the merge then.

**Review by ialidzhikov — APPROVED** [review]
/approve

**Comment by rfranzke** [comment]
Thanks!

**Pull request merged**